http/encoding.go

```diff
@@ -236,3 +245,45 @@ func SetContentType(w http.ResponseWriter, ct string) {
 	}
 	w.Header().Set("Content-Type", h+suffix)
 }
+
+func newTextHtmlEncoder(w io.Writer) Encoder {
+	return &textHtmlEncoder{w}
+}
+
+type textHtmlEncoder struct {
+	w io.Writer
+}
+
+func (e *textHtmlEncoder) Encode(v interface{}) error {
+	var b []byte
+
+	strptr, isstrptr := v.(*string)
```

Comment:
We should protect against panics here and return an error if v is not a string. Do we really need to support pointers? Also we should support slices of bytes.

Reply:
If you have a design that produces a result like
```go
Result(func() {
  Attribute("content", String, "The html content", func() {
    Example("<html>...</html>")
  })
  // .
  // .
})
```
where the "content" field is optional and you specify this field for the response body like
```go
HTTP(func() {
  GET("")
  Response(StatusOK, func() {
    Body("content")
    ContentType("text/html")
  })
})
```
I've seen the encoder called with a string pointer.

Reply:
OK makes sense, thanks.